clustered/integration-test/src/test/java/org/ehcache/clustered/reconnect/CacheStateRepositoryReconnectTest.java

```diff
@@ -0,0 +1,196 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.reconnect;
+
+import org.ehcache.Cache;
+import org.ehcache.PersistentCacheManager;
+import org.ehcache.clustered.client.config.builders.ClusteredResourcePoolBuilder;
+import org.ehcache.clustered.client.config.builders.ClusteringServiceConfigurationBuilder;
+import org.ehcache.clustered.client.internal.store.ReconnectInProgressException;
+import org.ehcache.clustered.util.TCPProxyManager;
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.builders.CacheConfigurationBuilder;
+import org.ehcache.config.builders.CacheManagerBuilder;
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.config.units.MemoryUnit;
+import org.junit.Test;
+import org.junit.BeforeClass;
+import org.junit.AfterClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.terracotta.testing.rules.Cluster;
+import org.terracotta.utilities.test.rules.TestRetryer;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static java.time.Duration.ofSeconds;
+import static org.ehcache.testing.StandardCluster.clusterPath;
+import static org.ehcache.testing.StandardCluster.leaseLength;
+import static org.ehcache.testing.StandardCluster.newCluster;
+import static org.ehcache.testing.StandardCluster.offheapResource;
+import static org.ehcache.testing.StandardTimeouts.eventually;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.terracotta.utilities.test.rules.TestRetryer.OutputIs.CLASS_RULE;
+import static org.terracotta.utilities.test.rules.TestRetryer.tryValues;
+
+public class CacheStateRepositoryReconnectTest {
+  private static TCPProxyManager proxyManager;
+  private static PersistentCacheManager cacheManager;
+
+  private static CacheConfiguration<NodeKey, NodeVal> config = CacheConfigurationBuilder.newCacheConfigurationBuilder(NodeKey.class, NodeVal.class, ResourcePoolsBuilder.newResourcePoolsBuilder().with(ClusteredResourcePoolBuilder.clusteredDedicated("primary-server-resource", 1, MemoryUnit.MB))).withResilienceStrategy(new ThrowingResiliencyStrategy<>()).build();
+
+  @ClassRule
+  @Rule
+  public static final TestRetryer<Duration, Cluster> CLUSTER = tryValues(ofSeconds(5)).map(leaseLength -> newCluster().in(clusterPath()).withServiceFragment(offheapResource("primary-server-resource", 64) + leaseLength(leaseLength)).build()).outputIs(CLASS_RULE);
```

Comment:
This line is more than 260 characters long -- it should be split apart so the lines are between 80 and 120 characters long.